WIP NT Research Facility

DeadmonWonderland opened the pull request:
Adds a NT Research Facility. Has a load of features planned that I'll update with as time goes on.

:cl: Deadmon
rscadd: Adds a NT Research Facility.
:cl:

HeyBanditoz commented:
[Here's a screenshot since the map diff doesn't seem to load it](https://cdn.discordapp.com/attachments/492799900145418270/715830712560779284/test.png)

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A isolator, it appears to be off."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "isolator_off"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An isolator, the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "isolator"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An isolator, there is a sample inserted and the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "isolator_processing"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An isolator, there is a sample inserted and the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "isolator_in"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An analyser, there is a sample inserted and the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "analyser_processing"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An analyser, it is currently sitting idle."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "analyser"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An incubator, it is currently idle."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "incubator"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "An incubator, you can see a set of samples inside."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "incubator_on"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A centrifuge, it is currently idle."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "centrifuge"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "Some sort of scanner, it appears there is someone inside as data streams off the console."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "scanner_1"
+	anchored = "1"
```
```suggestion
	anchored = TRUE

```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "Some sort of scanner, you're not sure what it does."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "scanner_0"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A console, the glass is smashed."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "broken"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A console, it's not functioning."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "c_unpowered"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A console, the information displayed is far beyond what you can understand."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "scannerconsole"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A console, you're not sure what the information displayed means."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "dna_computer"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A cryotube, a strange liquid bubbles inside the glass."
+	icon = 'icons/obj/cryogenics.dmi'
+	icon_state = "cell-on"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A cryotube, a figure floats inside, they do not appear to be conscious."
+	icon = 'icons/obj/cryogenics.dmi'
+	icon_state = "cell-occupied"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A cryotube, it is not on."
+	icon = 'icons/obj/cryogenics.dmi'
+	icon_state = "cell-off"
+	anchored = "1"  
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A centrifuge, the glass is smashed open."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "centrifugeb"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+	desc = "A centrifuge, it is currently processing a sample."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "centrifuge_moving"
+	anchored = "1"
```
```suggestion
	anchored = TRUE
```

Spookerton commented on maps/away/umb_zh/zhenghulab.dm:
```diff
@@ -0,0 +1,53 @@
+///////////////////////////////////crew and prisoners
+
+////////////////////////////Notes and papers
+/obj/item/weapon/paper/lar_maria/note_1
```
```suggestion
/obj/item/weapon/paper/umbrella/note_1
```

Spookerton commented on code/game/objects/structures/signs.dm:
```diff
@@ -60,6 +60,11 @@
 		qdel(src)
 	else ..()
 
+/obj/item/sign/ntplague
+	name = "Plague"
+	desc = "A plague, dedicated to the founder of NT."
+	icon_state = 'lightplaquealt'
```
```suggestion
/obj/item/sign/ntplaque
	name = "Plaque"
	desc = "A plaque, dedicated to the founder of NanoTrasen."
	icon_state = "lightplaquealt"
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/iso
+	name = "Isolator"
+	desc = "An isolator, the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/iso_on
+	name = "Isolator"
+	desc = "An isolator, there is a sample inserted and the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/iso_sample
+	name = "Isolator"
+	desc = "An isolator, there is a sample inserted and the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/analyser
+	name = "Analyser"
+	desc = "An analyser, there is a sample inserted and the screen has a series of data streams that you can't quite understand."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/analyser_off
+	name = "Analyser"
+	desc = "An analyser, it is currently sitting idle."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/incubator_off
+	name = "Incubator"
+	desc = "An incubator, it is currently idle."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/incubator_on
+	name = "Incubator"
+	desc = "An incubator, you can see a set of samples inside."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/centrifuge
+	name = "Centrifuge"
+	desc = "A centrifuge, it is currently idle."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/centrifuge_on
+	name = "Centrifuge"
+	desc = "A centrifuge, it is currently processing a sample."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on code/game/objects/structures/virologyprops.dm:
```diff
@@ -0,0 +1,139 @@
+/obj/structure/virology/centrifuge_broken
+	name = "Centrifuge"
+	desc = "A centrifuge, the glass is smashed open."
+	icon = 'icons/obj/virology.dmi'
```
```suggestion
```

Spookerton commented on maps/away/umb_zh/zhenghulab.dm:
```diff
@@ -0,0 +1,53 @@
+/obj/item/weapon/paper/lar_maria/note_1
+	name = "paper note"
+	info = {"
+			<center><b><font color='green'>Zeng-Hu Pharmaceuticals</font></b></center>
```
```suggestion
			<center><b><font color='green'>NanoTrasen</font></b></center>
```

Spookerton commented on maps/away/umb_zh/zhenghulab.dm:
```diff
@@ -0,0 +1,53 @@
+	icon_state = "object"
+	known = 0
+
+/datum/map_template/ruin/away_site/umbrellant
+	name = "NT Research Facility"
+	id = "awaysite_umbrellant"
+	description = "A bioweapons facility located on a strange planet."
+	suffixes = list("umb_zh/zhenghulab-1.dmm", "umb_zh/zhenghulab-2.dmm")
```
```suggestion
/datum/map_template/ruin/away_site/umbrella
	name = "NanoTrasen Research Facility"
	id = "awaysite_umbrella"
	description = "A bioweapons facility located on a strange planet."
	suffixes = list("umbrella/umbrella-1.dmm", "umbrella/umbrella-2.dmm")
```

Spookerton commented on maps/away/umb_zh/zhenghulab.dm:
```diff
@@ -0,0 +1,53 @@
+/obj/effect/overmap/visitable/sector/umbrellant
+	name = "Unknown Research Facility"
```
```suggestion
/obj/effect/overmap/visitable/sector/umbrella
	name = "Research Facility"
```

Spookerton commented on code/game/objects/structures/signs.dm:
```diff
@@ -74,6 +79,14 @@
 /obj/structure/sign/double/map/right
 	icon_state = "map-right"
 
+/obj/structure/sign/double/map/eternisright
+	desc = "A framed map of the Eternis Research Facility"
+	icon_state = "mapeternis-left
+
+/obj/structure/sign/double/map/eternisleft
+	desc = "A framed map of the Eternis Research Facility"
+	icon_state = "mapeternis-right"
```
```suggestion
/obj/structure/sign/double/map/umbrella_right
	desc = "A framed map of the Umbrella Research Facility"
	icon_state = "mapeternis-left

/obj/structure/sign/double/map/umbrella_left
	desc = "A framed map of the Umbrella Research Facility"
	icon_state = "mapeternis-right"
```

Spookerton commented on maps/away/umb_zh/zhenghulab-1.dmm:
```diff
@@ -0,0 +1,16458 @@
+/turf/simulated/floor/tiled/dark,
+/area/space)
+"RN" = (
+/obj/structure/sign/double/map/eternisleft{
```
```suggestion
/obj/structure/sign/double/map/umbrella_left{
```

Spookerton commented on maps/away/umb_zh/zhenghulab-1.dmm:
```diff
@@ -0,0 +1,16458 @@
+/turf/simulated/floor/tiled/white,
+/area/space)
+"Wo" = (
+/obj/structure/sign/double/map/eternisright{
```
```suggestion
/obj/structure/sign/double/map/umbrella_right{
```

Spookerton commented:
Also requires changes to pathing & file names:

`maps/away/umb_zh/` => `maps/away/umbrella`
`maps/away/umb_zh/zhenghulab.dm` => `maps/away/umbrella/umbrella.dm`
`maps/away/umb_zh/zhenghulab-2.dmm` =>`maps/away/umbrella/umbrella-1.dmm`
`maps/away/umb_zh/zhenghulab-2.dmm` =>`maps/away/umbrella/umbrella-2.dmm`

And replacement of two other paths in maps:

`/obj/item/weapon/paper/lar_maria/note_1` => `/obj/item/weapon/paper/umbrella/note_1`
`/obj/item/sign/ntplague` => `/obj/item/sign/ntplaque`

DeadmonWonderland committed:
NT Research and Development

A NT Research facility
Aaffa
affa
more changes
Update code/game/objects/structures/virologyprops.dm

Co-authored-by: Spookerton <Spookerton@users.noreply.github.com>
Update code/game/objects/structures/virologyprops.dm

Co-authored-by: Spookerton <Spookerton@users.noreply.github.com>
Update code/game/objects/structures/virologyprops.dm

Co-authored-by: Spookerton <Spookerton@users.noreply.github.com>
Update code/game/objects/structures/virologyprops.dm

Co-authored-by: Spookerton <Spookerton@users.noreply.github.com>
Update code/game/objects/structures/virologyprops.dm

Co-authored-by: Spookerton <Spookerton@users.noreply.github.com>

DeadmonWonderland commented:
On hold for a hot minute while I work on another project for someone else

Spookerton commented:
Reopen this if/when you intend to revisit it.
